src/include/gurt/debug.h

```diff
@@ -325,6 +325,14 @@ int d_register_alt_assert(void (*alt_assert)(const int, const char*,
 		fflush(stdout);						\
 	} while (0)
 
+#ifdef DAOS_ASSERT_DISABLE
+
+#define D_ASSERT(cond)
+
+#define D_ASSERTF(cond, fmt, ...)
```

Comment:
Instead of changing every D_ASSERT() call to add "{ }", why don't we define it as "#define D_ASSERT(cond) do { } while (0)"? Other ongoing PR may still use the coding style of "if (cond) D_ASSERT()", they would be conflicting with this PR, and such conflicts is hard to be detected.

Reply:
I've checked that and it would actually work.  I found the extra `{}` useful though to make it clearer that D_ASSERT() may be a no-op.  I'm not worried about conflicts, as I said this PR just adds a new compile option which works today as I've tested it but no-where does it enable that option, I'll have to address that in a future PR.